internal/template/template_test.go

```diff
@@ -91,6 +92,40 @@ func TestMergeData(t *testing.T) {
 	}
 }
 
+func TestCopyData(t *testing.T) {
+	nested := map[string]interface{}{
+		"c": []int{1, 2, 3},
+	}
+	addrNested := fmt.Sprintf("%p", nested)
+	cases := []struct {
+		name  string
+		input map[string]interface{}
+		want  map[string]interface{}
+	}{
+		{
+			name: "copy nested map",
+			input: map[string]interface{}{
+				"a": nested,
+			},
+			want: map[string]interface{}{
+				"a": nested,
+			},
+		},
+	}
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			copy := CopyData(tc.input)
+			addrCopy := fmt.Sprintf("%p", copy["a"])
```

Comment:
no need to create a var for this if the value is just used once

Reply:
I removed using addresses from the test. Instead, I test for side-effect: compare copy with original before and after an external modification. If deep copy fails then objects do not differ after a modification.